src/test/java/plugins/GitLabPluginTest.java

```diff
@@ -0,0 +1,323 @@
+        return privateTokenUser;
+    }
+    @Before
+    public void init() throws InterruptedException, IOException {
+        container = gitLabServer.get();
+        host = container.host();
+        port = container.sshPort();
+        container.waitForReady(this);
+
+        // create an admin user
+        privateTokenAdmin = container.createUserToken(adminUserName, "arandompassword12#", "testadmin@invalid.test", "true");
+
+        // create another user
+        privateTokenUser = container.createUserToken(userName, "passwordforsimpleuser12#", "testsimple@invalid.test", "false");
+    }
+
+    @Test
+    public void testGitLabMultibranchPipeline() throws IOException, GitLabApiException {
+        createRepo();
+
+        // initialize GitLabApi and Project
+        GitLabApi gitlabapi = new GitLabApi(container.getHttpUrl().toString(), getPrivateTokenAdmin());
+        ProjectApi projApi = new ProjectApi(gitlabapi);
+        Project project = projApi.getProjects().stream().filter((proj -> repoName.equals(proj.getName()))).findAny().orElse(null);
+
+        // create 2 new branches
+        createBranch(gitlabapi, project, firstBranch, mainBranch);
+        createBranch(gitlabapi, project, secondBranch, mainBranch);
+
+        // a branch which is intended to fail
+        createBranch(gitlabapi, project, failedJob, mainBranch);
+
+        // add Jenkinsfile on these branches
+        addFile(gitlabapi, project, mainBranch, JenkinsfileContent);
+        addFile(gitlabapi, project, firstBranch, JenkinsfileContent);
+        addFile(gitlabapi, project, secondBranch, JenkinsfileContent);
+
+        // add a broken Jekinsfile to the failedjob
+        addFile(gitlabapi, project, failedJob, brokenJenkinsfile);
+
+        // create a merge request
+        createMergeRequest(gitlabapi, project, secondBranch, mainBranch, "test_mr");
+
+        createGitLabToken(privateTokenAdmin, "GitLab Personal Access Token");
+        configureGitLabServer();
+
+        final WorkflowMultiBranchJob multibranchJob = jenkins.jobs.create(WorkflowMultiBranchJob.class);
+        this.configureJobWithGitLabBranchSource(multibranchJob, adminUserName, repoName);
+        multibranchJob.save();
+
+        multibranchJob.waitForBranchIndexingFinished((int)time.seconds(20));
+        this.assertBranchIndexing(multibranchJob);
+
+        final WorkflowJob successJob1 = multibranchJob.getJob(mainBranch);
+        final WorkflowJob successJob2 = multibranchJob.getJob(firstBranch);
+        final WorkflowJob successJob3 = multibranchJob.getJob("MR-1");
+        final WorkflowJob failureJob = multibranchJob.getJob(failedJob);
+
+        this.assertExistAndResult(successJob1, true);
+        this.assertExistAndResult(successJob2, true);
+        this.assertExistAndResult(successJob3, true);
+        this.assertExistAndResult(failureJob, false);
+
+        // delete the repo when finished
+        container.deleteRepo(getPrivateTokenAdmin(), repoName);
```

Comment:
This should be done in a `@After` method or in a `try { ... } finally {...}` block because if any of the asserts above fails then the repo won't be deleted otherwise.